src/components/suggest_input/suggest_input.js

```diff
@@ -0,0 +1,144 @@
+/* eslint-disable @typescript-eslint/no-unused-vars */
+import React, { Component } from 'react';
+import PropTypes from 'prop-types';
+import classNames from 'classnames';
+import { EuiFilterButton } from '../filter_group';
+import { EuiFieldText } from '../form';
+import { EuiToolTip } from '../tool_tip';
+import { EuiIcon } from '../icon';
+import { EuiPopover } from '../popover';
+
+export class EuiSuggestInput extends Component {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      showSuggestions: false,
+      value: '',
+      menuWidth: null,
+    };
+  }
+
+  // onFieldFocus() {
+  //   this.setState({
+  //     showSuggestions: true,
+  //   });
+  // }
+
+  onFieldBlur() {
+    this.setState({
+      showSuggestions: false,
+    });
+  }
+
+  setPopoverRef = ref => {
+    this.inputRef = ref;
+  };
+
+  onFieldChange(e) {
+    this.setState({
+      value: e.target.value,
+      showSuggestions: e.target.value === '' ? false : true,
+    });
+
+    const focusSelected = () => {
+      requestAnimationFrame(() => {
+        this.setState({
+          menuWidth: this.inputRef.getBoundingClientRect().width + 50,
+        });
+      });
+    };
+
+    requestAnimationFrame(focusSelected);
+  }
+
+  render() {
+    const {
+      className,
+      status,
+      label,
+      prefix,
+      suggestions,
+      ...rest
+    } = this.props;
+
+    const statusMap = {
+      notYetSaved: {
+        icon: 'dot',
+        color: '#DD0A73',
+        tooltip: "You've made changes to this saved query. Click to save them.",
+      },
+      saved: {
+        icon: 'checkInCircleFilled',
+        color: 'secondary',
+      },
+      noNewChanges: {
+        icon: '',
+        color: 'secondary',
+      },
+    };
+
+    let icon;
+    let color;
+    let tooltip;
+
+    if (statusMap[status]) {
+      icon = statusMap[status].icon;
+      color = statusMap[status].color;
+      tooltip = statusMap[status].tooltip;
+    }
+    const statusElement = (
+      <EuiToolTip position="left" content={tooltip}>
+        {status === 'isLoading' ? (
+          <span className="euiLoadingSpinner euiLoadingSpinner--medium" />
+        ) : (
+          <div className="statusIcon">
+            <EuiIcon color={color} type={icon} />
+            <span className="statusLabel">{label}</span>
+          </div>
+        )}
+      </EuiToolTip>
+    );
+    const classes = classNames('euiSuggestInput', className);
+
+    return (
+      <div className={classes}>
+        <EuiFieldText
+          value={this.state.value}
+          fullWidth
+          prepend={prefix}
+          append={statusElement}
```

Comment:
In order to also allow the consumer to pass down any number of `append` items, you can just put your element to the beginning of the append array like:

```suggestion
          append={append ? append.unshift(statusElement) : statusElement}
```